src/pybind/mgr/orchestrator.py

```diff
@@ -28,40 +29,72 @@
 except ImportError:
     T, G = object, object
 
-def split_host(host):
-    """
-    Split host into host and (optional) daemon name parts
-    e.g.,
-      "myhost=name" -> ('myhost', 'name')
-      "myhost" -> ('myhost', None)
-    """
-    # TODO: stricter validation
-    a = host.split('=', 1)
-    if len(a) == 1:
-        return (a[0], None)
-    else:
-        assert len(a) == 2
-        return tuple(a)
-
-def split_host_with_network(host):
+def parse_host_specs(host, require_network=True):
     """
     Split host into host, network, and (optional) daemon name parts.  The network
     part can be an IP, CIDR, or ceph addrvec like
     '[v2:1.2.3.4:3300,v1:1.2.3.4:6789]'.
     e.g.,
-      "myhost:1.2.3.4=name" -> ('myhost', '1.2.3.4', 'name')
-      "myhost:1.2.3.0/24" -> ('myhost', '1.2.3.0/24', None)
+      "myhost"
+      "myhost=name"
+      "myhost:1.2.3.4"
+      "myhost:1.2.3.4=name"
+      "myhost:1.2.3.0/24"
+      "myhost:1.2.3.4/24=name"
+      "myhost:v2:1.2.3.4:3000/24=name"
+      "myhost:v2:1.2.3.4:3000,v1:1.2.3.4:6789/24=name"
```

Comment:
could you add a test for this in src/pybind/mgr/tests/test_orchestrator.py ?

Reply:
will do